### PR TITLE
List + Edit + Delete progress item

### PR DESCRIPTION
**Pull request by ghostp13409**
- Added Some Comments
- Added Item Class for Handling Common Fields
- Implemented ConsoleTables

Issues no:
#87 
#88 
#86

### CODE REVIEW

**Comment by ehharvey** [comment]
It's looking good! Apologies, I've been having some midnight work changes which have wrecked my sleep. But should be better today to give feedback!

Revisions I'm seeing so far are small! So should be not much more work I'm thinking

**Comment by ehharvey** [comment]
I reviewed this on my phone lol so just want to do some checks on my laptop tonight :) before submitting my review

File: app/models/Item.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            Description,
+            WorkItem // WorkItem(Id)
+        }
+
+        // Fields
+        public int Id { get; set; }
+
+        public DateTime CreatedAt { get; } = DateTime.Now;
+    }
+}
```
Comment (ehharvey) [inline app/models/Item.cs:21]: I really like this idea for an item, but it is out-of-scope so I am going to ask you to revert this.

Generally, we want to make sure our branches are focused on the work being asked. Otherwise, this leads to scope creep. It also increases the risk of error. I don't mean to ramble here, but do feel free to chat with me on Slack if you want to discuss how this can affect risk.

I had an issue at work the other day where a small (2-line) but unrelated change was added by a dev which casued some regressions.

File: app/CommandLineParser.cs [hunk 2]
```diff
@@ -62,6 +63,7 @@ class CommandLineParser : ICommandLineParser
         // Credits supports just the List verb.
         { Noun.Credits, new HashSet<Verb> { Verb.List } },
         { Noun.WorkItem, new HashSet<Verb> { Verb.List, Verb.Create, Verb.Edit, Verb.Delete } },
+        { Noun.Progress, new HashSet<Verb> { Verb.List, Verb.Create, Verb.Edit, Verb.Delete} },
```
Comment (ehharvey) [inline app/CommandLineParser.cs:66]: Since this issue only handles edits and delete, we should not include Create or List.

File: app/Credits.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 It was created to fulfill #16
 */
 
+using ConsoleTables;
```
Comment (ehharvey) [inline app/Credits.cs:6]: This also looks to me like a bit of scope creep. Can you revert this?

File: app/Progress.cs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            //    break;
+                
+            default:
+                throw new ArgumentException("Invalid Verb");
+        }
+    }
+
+    // Edit Progress
+    public Lms.Models.Progress Edit(string id, string field, string value) {
+        int parsed_id = -1;
```
Comment (ehharvey) [inline app/Progress.cs:169]: There's a concept called [Resource Acquisition is Initialization](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) which basically is the idea to set variable values at time of declaration (to avoid things like having to think of temp values).

I think it is A OK here :) just thought to share this in case you're interested!

File: app/WorkItem.cs [hunk 5]
```diff
@@ -1,17 +1,22 @@
+using ConsoleTables;
+using lms.models;
 using Lms;
 
-enum Field {
-    Title,
-    DueAt
-}
```
Comment (ehharvey) [inline app/WorkItem.cs:6]: Another case of scope creep unfortunately. I don't disagree that this could be a good change, but it is unrelated to the issues this branch is linked to. Can you revert?

**Review by ehharvey — CHANGES_REQUESTED** [review]

File: app/WorkItem.cs (hunk 5 above)
Comment (ghostp13409) [reply app/WorkItem.cs:6]: Thanks for the Feedback Emil! I will make these changes. However, i still have some confusion about Implementing Fields. Currently Field is a global enum, so i cannot create another enum for progresses. I created the extra Items class because of this. I am not sure how to implement the Field to progress. The one possible way is to just add the extra field names to this Field, which is in workItems. However, that seems a bit weird to access it from workItems.

Comment (ehharvey) [reply app/WorkItem.cs:6]: Hmm good point. My long-term thinking would be eventually moving WorkItem and Progress to their own namespace.

In the short-term, can you make the enum `ProgressField`. Leave the WorkItem `Field` the same for now, we can tackle that later :)

File: app/WorkItem.cs [hunk 6]
```diff
@@ -1,17 +1,28 @@
...
-
 class WorkItem : ICommand {
+
+    public enum Field
```
Comment (ehharvey) [inline app/WorkItem.cs:6]: Good idea!

File: app/models/WorkItem.cs [hunk 7]
```diff
@@ -1,15 +1,14 @@
+using System.Security.Cryptography;
```
Comment (ehharvey) [inline app/models/WorkItem.cs:1]: This looks like it's maybe a dependency that we should remove. NBD but we can tackle it later :)

**Review by ehharvey — APPROVED** [review]

**Comment by ehharvey** [comment]
Reverted, see #226
